builtin/providers/aws/resource_aws_elastic_transcoder_pipeline.go

```diff
@@ -0,0 +1,393 @@
+	elastictranscoderconn := meta.(*AWSClient).elastictranscoderconn
+
+	req := &elastictranscoder.UpdatePipelineInput{
+		Id: aws.String(d.Id()),
+	}
+
+	if d.HasChange("aws_kms_key_arn") {
+		req.AwsKmsKeyArn = getStringPtr(d, "aws_kms_key_arn")
+	}
+
+	if d.HasChange("content_config") {
+		req.ContentConfig = expandETPiplineOutputConfig(d, "content_config")
+	}
+
+	if d.HasChange("input_bucket") {
+		req.Role = getStringPtr(d, "input_bucket")
+	}
+
+	if d.HasChange("name") {
+		req.Name = getStringPtr(d, "name")
+	}
+
+	if d.HasChange("notifications") {
+		req.Notifications = expandETNotifications(d)
+	}
+
+	if d.HasChange("role") {
+		req.Role = getStringPtr(d, "role")
+	}
+
+	if d.HasChange("thumbnail_config") {
+		req.ThumbnailConfig = expandETPiplineOutputConfig(d, "thumbnail_config")
+	}
+
+	log.Printf("[DEBUG] Updating Elastic Transcoder Pipeline: %#v", req)
+	output, err := elastictranscoderconn.UpdatePipeline(req)
+	if err != nil {
+		return fmt.Errorf("Error updating Elastic Transcoder pipeline: %s", err)
+	}
+
+	for _, w := range output.Warnings {
+		log.Printf("[WARN] Elastic Transcoder Pipeline %s: %s", w.Code, w.Message)
+	}
+
+	return resourceAwsElasticTranscoderPipelineRead(d, meta)
+}
+
+func resourceAwsElasticTranscoderPipelineRead(d *schema.ResourceData, meta interface{}) error {
+	elastictranscoderconn := meta.(*AWSClient).elastictranscoderconn
+
+	resp, err := elastictranscoderconn.ReadPipeline(&elastictranscoder.ReadPipelineInput{
+		Id: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == "ResourceNotFoundException" {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	log.Printf("[DEBUG] Elastic Transcoder Pipeline Read response: %#v", resp)
+
+	pipeline := resp.Pipeline
+
+	d.Set("arn", *pipeline.Arn)
+
+	if arn := pipeline.AwsKmsKeyArn; arn != nil {
+		d.Set("aws_kms_key_arn", *arn)
+	}
+
+	if pipeline.ContentConfig != nil {
+		d.Set("content_config", flattenETPipelineOutputConfig(pipeline.ContentConfig))
+	}
+
+	d.Set("input_bucket", *pipeline.InputBucket)
+	d.Set("name", *pipeline.Name)
+
+	notifications := flattenETNotifications(pipeline.Notifications)
+	if notifications != nil {
+		d.Set("notifications", notifications)
+	}
+
+	if pipeline.OutputBucket != nil {
+		d.Set("output_bucket", *pipeline.OutputBucket)
+	}
+
+	d.Set("role", pipeline.Role)
+
+	if pipeline.ThumbnailConfig != nil {
+		d.Set("thumbnail_config", flattenETPipelineOutputConfig(pipeline.ThumbnailConfig))
+	}
+
+	return nil
+}
+
+func resourceAwsElasticTranscoderPipelineDelete(d *schema.ResourceData, meta interface{}) error {
+	elastictranscoderconn := meta.(*AWSClient).elastictranscoderconn
+
+	log.Printf("[DEBUG] Elastic Transcoder Delete Pipeline: %s", d.Id())
+	_, err := elastictranscoderconn.DeletePipeline(&elastictranscoder.DeletePipelineInput{
+		Id: aws.String(d.Id()),
+	})
+	if err != nil {
+		return fmt.Errorf("error deleting Elastic Transcoder Pipeline: %s", err)
```

Comment:
Should we prefix errors with `[ERR]`, or log statements? Don't we prefix the level in the log line anyway?